crates/solver/src/settlement_rater.rs

```diff
@@ -244,6 +245,23 @@ impl SettlementRating for SettlementRater {
             None => Score::Protocol(big_rational_to_u256(&objective_value).unwrap_or_default()),
         };
 
+        // recalculate score if success probability is provided
+        let score = match settlement.success_probability {
+            Some(success_probability) => {
+                match self
+                    .score_calculator
+                    .compute_score(&objective_value, success_probability)
+                {
+                    Ok(score) => score,
+                    Err(err) => {
+                        tracing::warn!(?err, "Failed to compute score with success probability");
+                        score
+                    }
+                }
+            }
+            None => score, // remove once success probability becomes mandatory
```

Comment:
Created issue: https://github.com/cowprotocol/services/issues/1821
Removed comment since this code will be inevitably touched once issue is implemented.